source/fundamentals/crud/read-operations/project.txt

```diff
@@ -0,0 +1,169 @@
+
+Project
+-------
+
+To specify whether fields in a document should return, pass an interface
+specifying the field and ``0`` or ``1`` to the ``SetProjection()``
+function of a read operation's options.
+
+Specify the options as the last parameter to the following read
+operation functions:
+
+- ``Find()``
+- ``FindOne()``
+- ``FindOneAndDelete()``
+- ``FindOneAndReplace()``
+- ``FindOneAndUpdate()``
+
+Exception
+~~~~~~~~~
+
+When specifying a projection, you can only specify to either include or
+exclude fields. This rule **does not** apply to the ``_id`` field. You
+can exclude the ``_id`` field even if you specified to include certain
+fields.
+
+By default, the driver includes the ``_id`` field. You must explicitly
+exclude the ``_id`` field if you do not want it returned. For an example
+of this scenario, see the :ref:`include a field section <go-include-projection>`.
+
+.. tip::
+
+   If you don't specify a projection, the read operation returns all
+   the fields in matched documents.
+
+Exclude a Field
+~~~~~~~~~~~~~~~
+
+Specify a ``0`` in the interface passed to the ``SetProjection()``
+function to exclude a field. For all fields you don't explicitly list
+in the interface, the driver includes them.
+
+Example
+```````
+
+The following example excludes the ``rating`` from the matched documents
+from the ``Find()`` function:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/projection.go
+   :language: go
+   :dedent:
+   :start-after: begin exclude projection
+   :end-before: end exclude projection
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   //results truncated
+   [{_id ObjectID("...")} {type Masala}]
+   [{_id ObjectID("...")} {type Assam}]
+   [{_id ObjectID("...")} {type Oolong}]
+   [{_id ObjectID("...")} {type Earl Grey}]
+   [{_id ObjectID("...")} {type English Breakfast}]
+
+Include a Field
+~~~~~~~~~~~~~~~
+
+Specify a ``1`` in the interface passed to the ``SetProjection()``
+function to include a field. For all fields you don't explicitly list
+in the interface, the driver excludes them.
+
+.. important::
+
+    You can exclude the ``_id`` field even if you specified to include
+    certain fields.
+
+.. _go-include-projection:
+
+Example
+```````
+
+The following example performs the following projection on the matched
+documents from the ``Find()`` function:
+
+- Include the ``type`` and ``rating`` field
+- Exclude the ``_id`` field
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/projection.go
+   :language: go
+   :dedent:
+   :start-after: begin include projection
+   :end-before: end include projection
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{type Masala} {rating 10}]
+   [{type Assam} {rating 5}]
+   [{type Oolong} {rating 7}]
+   [{type Earl Grey} {rating 8}]
+   [{type English Breakfast} {rating 5}]
+
+.. tip:: Using Aggregation 
```

Comment:
I would consider having this example stand on its own rather than in a tip box. I think it's a good example to include and should be treated the same as the others.